src/components/text-area.jsx

```diff
@@ -0,0 +1,54 @@
+var TextArea = React.createClass({
+    propTypes: {
+        value: React.PropTypes.string,
+        cols: React.PropTypes.number,
+        rows: React.PropTypes.number,
+        onChange: React.PropTypes.func.isRequired,
+        className: React.PropTypes.string,
+        onFocus: React.PropTypes.func,
+        onBlur: React.PropTypes.func
+    },
+
+    getDefaultProps: function() {
+        return {
+            value: ""
+        };
+    },
+
+    render: function() {
+        return <textarea
+            {...this.props}
```

Comment:
I think this "{...this.props}" together with the "{...this.props}" inside free-text-input.jsx  creates a loop